HDVF/include/CGAL/HDVF/Filtration_lower_star.h

```diff
@@ -0,0 +1,236 @@
+
+ The map is extended to cells of any dimension by setting, for a cell \f$\sigma\f$:
+ \f[\mathrm{deg}(\sigma) = \max_{\substack{v\in K_0\\v\text{ face of }\sigma}} \mathrm{deg}(v)
+ \f]
+
+ For geometric complexes, standard lower star filtrations are obtained by taking as a degree function the \f$x\f$, \f$y\f$ or \f$z\f$ coordinate of vertices. The image below illustrates such a filtration ((left) lower star filtration with a \f$z\f$ degree map on vertices, (right) lower star filtration with a \f$y\f$ degree map on vertices).
+
+ <img src="lower_star_filtration_z.png" align="center" width=15%/>
+ <img src="lower_star_filtration_y.png" align="center" width=15%/>
+
+ The `Filtration_lower_star` class provides constructors taking as input:
+ - either the vector of vertex degrees
+ - or a function mapping each vertex to its degree.
+
+ \cgalModels{Filtration}
+
+ \tparam ChainComplex a model of the `AbstractChainComplex` concept (type of the underlying chain complex).
+ \tparam Degree the scalar type of degrees.
+ */
+
+template <typename ChainComplex, typename Degree>
+class Filtration_lower_star : public Filtration_core<ChainComplex, Degree>
+{
+private:
+    /*! \brief Type of coefficients used to compute homology. */
+    typedef ChainComplex::Coefficient_ring Coefficient_ring;
+    /*! Type of parent filtration instance. */
+    typedef Filtration_core<ChainComplex, Degree> Filtration_parent;
+    /*! Type of value returned by the iterator. */
+    typedef Filtration_parent::Filtration_iter_value Filtration_iter_value;
+public:
+    /*!
+     * \brief Copy constructor.
+     *
+     * Builds a filtration by copy from another.
+     *
+     * \param[in] f An initial lower star filtration.
+     */
+    Filtration_lower_star(const Filtration_lower_star& f) : Filtration_parent(f) {}
+
+    /*! \brief Constructor from vertex degrees.
+     *
+     * The constructor computes all cells degrees as the minimum of the degrees of their vertices and sorts all the cells of the complex to fulfill the filtration ordering constraints.
+     *
+     * \param[in] K Constant reference to the underlying complex.
+     * \param[in] deg Vector of vertex degrees.
+     */
+    Filtration_lower_star(const ChainComplex& K, const std::vector<Degree>& deg) : Filtration_parent(K)
+    {
+        star_filtration(deg);
+    }
+
+    /*! \brief Constructor from a function mapping vertices to degrees.
+     *
+     * The constructor computes all cells degrees as the minimum of the degrees of their vertices (obtained through `deg_fun`) and sorts all the cells of the complex to fulfill the filtration ordering constraints.
+     *
+     * \param[in] K Constant reference to the underlying complex.
+     * \param[in] deg_fun Function mapping vertices of `K` to their degree.
+     */
+    Filtration_lower_star(const ChainComplex& K, std::function<Degree(size_t)>& deg_fun) : Filtration_parent(K)
```

Comment:
The parameter `deg_fun`  must be renamed as otherwise there is a doxygen link. See [here](https://cgal.github.io/9004/v1/HDVF/classCGAL_1_1Homological__discrete__vector__field_1_1Filtration__lower__star.html#abd5d7eac3c5509887aa5bcbf32ab2abd)